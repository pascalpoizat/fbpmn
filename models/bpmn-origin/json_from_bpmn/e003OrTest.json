{
    "isInterrupt": {},
    "catN": {
        "Process_": "Process",
        "Or1": "OrGateway",
        "AT2": "AbstractTask",
        "Xor2": "XorGateway",
        "AT5": "AbstractTask",
        "NSE": "NoneStartEvent",
        "AT4": "AbstractTask",
        "Xor1": "XorGateway",
        "AT1": "AbstractTask",
        "Or2": "OrGateway",
        "Xor0": "XorGateway",
        "AT3": "AbstractTask",
        "Xor3": "XorGateway",
        "NEE": "NoneEndEvent"
    },
    "catE": {
        "e1": "NormalSequenceFlow",
        "e10": "NormalSequenceFlow",
        "e7": "NormalSequenceFlow",
        "e11": "NormalSequenceFlow",
        "e0": "NormalSequenceFlow",
        "e6": "NormalSequenceFlow",
        "e3": "ConditionalSequenceFlow",
        "e12": "DefaultSequenceFlow",
        "e9": "ConditionalSequenceFlow",
        "e13": "ConditionalSequenceFlow",
        "e2": "NormalSequenceFlow",
        "e8": "DefaultSequenceFlow",
        "e14": "NormalSequenceFlow",
        "e5": "NormalSequenceFlow",
        "e4": "ConditionalSequenceFlow",
        "e15": "DefaultSequenceFlow"
    },
    "containN": {
        "Process_": [
            "AT1",
            "Or1",
            "Xor1",
            "AT3",
            "AT2",
            "Xor2",
            "Or2",
            "Xor3",
            "AT4",
            "AT5",
            "NEE",
            "NSE",
            "Xor0"
        ]
    },
    "messageE": {},
    "timeInformation": {},
    "targetE": {
        "e1": "AT1",
        "e10": "AT4",
        "e7": "Xor2",
        "e11": "Xor3",
        "e0": "Xor0",
        "e6": "AT2",
        "e3": "Xor1",
        "e12": "Or2",
        "e9": "Or2",
        "e13": "AT5",
        "e2": "Or1",
        "e8": "Xor1",
        "e14": "NEE",
        "e5": "Or2",
        "e4": "AT3",
        "e15": "Xor0"
    },
    "name": "e003OrTest",
    "attached": {},
    "containE": {
        "Process_": [
            "e15",
            "e1",
            "e14",
            "e13",
            "e12",
            "e11",
            "e10",
            "e5",
            "e9",
            "e8",
            "e7",
            "e6",
            "e4",
            "e3",
            "e2",
            "e0"
        ]
    },
    "edges": [
        "e15",
        "e1",
        "e14",
        "e13",
        "e12",
        "e11",
        "e10",
        "e5",
        "e9",
        "e8",
        "e7",
        "e6",
        "e4",
        "e3",
        "e2",
        "e0"
    ],
    "messages": [],
    "sourceE": {
        "e1": "Xor0",
        "e10": "Or2",
        "e7": "AT2",
        "e11": "AT4",
        "e0": "NSE",
        "e6": "Xor1",
        "e3": "Or1",
        "e12": "Xor3",
        "e9": "Xor2",
        "e13": "Xor3",
        "e2": "AT1",
        "e8": "Xor2",
        "e14": "AT5",
        "e5": "AT3",
        "e4": "Or1",
        "e15": "Or1"
    },
    "nodes": [
        "Process_",
        "AT1",
        "Or1",
        "Xor1",
        "AT3",
        "AT2",
        "Xor2",
        "Or2",
        "Xor3",
        "AT4",
        "AT5",
        "NEE",
        "NSE",
        "Xor0"
    ],
    "nameN": {
        "AT2": "AT2",
        "AT5": "AT5",
        "AT4": "AT4",
        "AT1": "AT1",
        "AT3": "AT3"
    }
}