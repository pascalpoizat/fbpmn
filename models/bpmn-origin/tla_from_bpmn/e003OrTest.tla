---------------- MODULE e003OrTest ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Process_" :> {  }

ContainRel ==
  "Process_" :> { "AT1", "Or1", "Xor1", "AT3", "AT2", "Xor2", "Or2", "Xor3", "AT4", "AT5", "NEE", "NSE", "Xor0" }

Node == { "Process_", "AT1", "Or1", "Xor1", "AT3", "AT2", "Xor2", "Or2", "Xor3", "AT4", "AT5", "NEE", "NSE", "Xor0" }

Edge == { "e15", "e1", "e14", "e13", "e12", "e11", "e10", "e5", "e9", "e8", "e7", "e6", "e4", "e3", "e2", "e0" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "e15" :> "Or1"
@@ "e1" :> "Xor0"
@@ "e14" :> "AT5"
@@ "e13" :> "Xor3"
@@ "e12" :> "Xor3"
@@ "e11" :> "AT4"
@@ "e10" :> "Or2"
@@ "e5" :> "AT3"
@@ "e9" :> "Xor2"
@@ "e8" :> "Xor2"
@@ "e7" :> "AT2"
@@ "e6" :> "Xor1"
@@ "e4" :> "Or1"
@@ "e3" :> "Or1"
@@ "e2" :> "AT1"
@@ "e0" :> "NSE"

target ==
   "e15" :> "Xor0"
@@ "e1" :> "AT1"
@@ "e14" :> "NEE"
@@ "e13" :> "AT5"
@@ "e12" :> "Or2"
@@ "e11" :> "Xor3"
@@ "e10" :> "AT4"
@@ "e5" :> "Or2"
@@ "e9" :> "Or2"
@@ "e8" :> "Xor1"
@@ "e7" :> "Xor2"
@@ "e6" :> "AT2"
@@ "e4" :> "AT3"
@@ "e3" :> "Xor1"
@@ "e2" :> "Or1"
@@ "e0" :> "Xor0"

CatN ==
   "Process_" :> Process
@@ "AT1" :> AbstractTask
@@ "Or1" :> InclusiveOr
@@ "Xor1" :> ExclusiveOr
@@ "AT3" :> AbstractTask
@@ "AT2" :> AbstractTask
@@ "Xor2" :> ExclusiveOr
@@ "Or2" :> InclusiveOr
@@ "Xor3" :> ExclusiveOr
@@ "AT4" :> AbstractTask
@@ "AT5" :> AbstractTask
@@ "NEE" :> NoneEndEvent
@@ "NSE" :> NoneStartEvent
@@ "Xor0" :> ExclusiveOr

CatE ==
   "e15" :> DefaultSeqFlow
@@ "e1" :> NormalSeqFlow
@@ "e14" :> NormalSeqFlow
@@ "e13" :> ConditionalSeqFlow
@@ "e12" :> DefaultSeqFlow
@@ "e11" :> NormalSeqFlow
@@ "e10" :> NormalSeqFlow
@@ "e5" :> NormalSeqFlow
@@ "e9" :> ConditionalSeqFlow
@@ "e8" :> DefaultSeqFlow
@@ "e7" :> NormalSeqFlow
@@ "e6" :> NormalSeqFlow
@@ "e4" :> ConditionalSeqFlow
@@ "e3" :> ConditionalSeqFlow
@@ "e2" :> NormalSeqFlow
@@ "e0" :> NormalSeqFlow

LOCAL preEdges ==
   <<"Or1", "e2">> :> {"e0", "e1", "e15"}
@@ <<"Or2", "e5">> :> {"e0", "e1", "e15", "e2", "e4"}
@@ <<"Or2", "e9">> :> {"e0", "e1", "e15", "e2", "e3", "e6", "e7", "e8"}
@@ <<"Or2", "e12">> :> {"e10", "e11"}
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

