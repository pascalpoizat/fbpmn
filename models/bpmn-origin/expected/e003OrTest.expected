Working in /tmp/e003OrTest.PRT38 with 1 worker(s)
[92mtransformation done[0m
<<"Processes=", 1, "Nodes=", 14, "Gateway=", 6, "SF=", 16, "MF=", 0>>
---------- Network01Bag ----------
[X] Prop01Type
     states=41 trans=59 depth=15
[X] Prop02Safe
     states=41 trans=59 depth=15
[X] Prop03Sound
     states=41 trans=59 depth=15
[X] Prop04MsgSound
     states=41 trans=59 depth=15
---------- Network02FifoPair ----------
[X] Prop01Type
     states=41 trans=59 depth=15
[X] Prop02Safe
     states=41 trans=59 depth=15
[X] Prop03Sound
     states=41 trans=59 depth=15
[X] Prop04MsgSound
     states=41 trans=59 depth=15
---------- Network03Causal ----------
[X] Prop01Type
     states=41 trans=59 depth=15
[X] Prop02Safe
     states=41 trans=59 depth=15
[X] Prop03Sound
     states=41 trans=59 depth=15
[X] Prop04MsgSound
     states=41 trans=59 depth=15
---------- Network04Inbox ----------
[X] Prop01Type
     states=41 trans=59 depth=15
[X] Prop02Safe
     states=41 trans=59 depth=15
[X] Prop03Sound
     states=41 trans=59 depth=15
[X] Prop04MsgSound
     states=41 trans=59 depth=15
---------- Network05Outbox ----------
[X] Prop01Type
     states=41 trans=59 depth=15
[X] Prop02Safe
     states=41 trans=59 depth=15
[X] Prop03Sound
     states=41 trans=59 depth=15
[X] Prop04MsgSound
     states=41 trans=59 depth=15
---------- Network06Fifo ----------
[X] Prop01Type
     states=41 trans=59 depth=15
[X] Prop02Safe
     states=41 trans=59 depth=15
[X] Prop03Sound
     states=41 trans=59 depth=15
[X] Prop04MsgSound
     states=41 trans=59 depth=15
---------- Network07RSC ----------
[X] Prop01Type
     states=41 trans=59 depth=15
[X] Prop02Safe
     states=41 trans=59 depth=15
[X] Prop03Sound
     states=41 trans=59 depth=15
[X] Prop04MsgSound
     states=41 trans=59 depth=15
done.
